dsls/sql/src/main/java/org/apache/beam/dsls/sql/BeamSql.java

```diff
@@ -151,15 +152,20 @@ private void registerTables(PCollectionTuple input){
    */
   private static class SimpleQueryTransform
       extends PTransform<PCollection<BeamSqlRow>, PCollection<BeamSqlRow>> {
+    private static final String PCOLLECTION_TABLE_NAME = "PCOLLECTION";
     BeamSqlEnv sqlEnv = new BeamSqlEnv();
     private String sqlQuery;
 
     public SimpleQueryTransform(String sqlQuery) {
       this.sqlQuery = sqlQuery;
+      validateQuery();
     }
 
-    @Override
-    public PCollection<BeamSqlRow> expand(PCollection<BeamSqlRow> input) {
+    // public SimpleQueryTransform withUdf(String udfName){
+    // throw new UnsupportedOperationException("Pending for UDF support");
+    // }
```

Comment:
Best not to check in commented-out code. Since I have no other requests for changes, I'll remove as part of merging.